lib/template/macros.c

```diff
@@ -515,97 +515,96 @@ log_macro_expand(GString *result, gint id, gboolean escape, const LogTemplateOpt
        *   message specific timezone, if one is specified
        *   local timezone
        */
-      zone_ofs = (opts->time_zone_info[tz] != NULL ? time_zone_info_get_offset(opts->time_zone_info[tz],
-                  stamp->tv_sec) : stamp->zone_offset);
-      if (zone_ofs == -1)
-        zone_ofs = stamp->zone_offset;
-
-      t = stamp->tv_sec + zone_ofs;
-
-      cached_gmtime(&t, &tm_storage);
-      tm  = &tm_storage;
+      WallClockTime wct;
 
+      convert_unix_time_to_wall_clock_time_with_tz_override(&wct, stamp,
+                                                            time_zone_info_get_offset(opts->time_zone_info[tz], stamp->ut_sec));
```

Comment:
The last parameter looks strange for me. According to the `convert_unix_time_to_wall_clock_time_with_tz_override` signature, that should be a `gmtoff_override` which should be related to timezone. But stamp->ut_sec is the calendar time itself. Shouldn't that be zero?

Reply:
The only use for stamp->ut_sec is the parameter for time_zone_info_get_offset () which will return a gmtoff value.

This actually forces a timestamp to be in a specific timezone, with its associated daylight saving rules.

Reply:
Looking at `convert_unix_time_to_wall_clock_time_with_tz_override` I found something strange:

In this call it has 4 parameters:
https://github.com/bazsi/syslog-ng/blob/2c17c0620f87a696aa4c2282df3ff286db8edd4b/lib/template/macros.c#L520

In the definition it has only 3:
https://github.com/bazsi/syslog-ng/blob/2c17c0620f87a696aa4c2282df3ff286db8edd4b/lib/timeutils/conv.c#L98

Which is strange because the tests should have detected that. Do I miss something?

Reply:
Oh right sorry!